official/resnet/resnet_run_loop.py

```diff
@@ -355,21 +370,35 @@ def resnet_main(flags, model_function, input_function, shape=None):
       allow_soft_placement=True)
 
   # Set up a RunConfig to save checkpoint and set session config.
-  run_config = tf.estimator.RunConfig().replace(save_checkpoints_secs=1e9,
-                                                session_config=session_config)
+  if not flags.use_distribution_strategy:
+    run_config = tf.estimator.RunConfig().replace(
+        save_checkpoints_secs=1e9, session_config=session_config)
+  else:
+    if flags.gpus_for_distribution_strategy == 0:
+      distribution = one_device_strategy.OneDeviceStrategy(
+          'device:CPU:0')
+    elif flags.gpus_for_distribution_strategy == 1:
+      distribution = one_device_strategy.OneDeviceStrategy(
+          'device:GPU:0')
+    else:
+      distribution = mirrored_strategy.MirroredStrategy(
+          num_gpus=flags.gpus_for_distribution_strategy
+      )
+    run_config = tf.estimator.RunConfig(distribute=distribution).replace(
+        save_checkpoints_secs=1e9, session_config=session_config)
+
   classifier = tf.estimator.Estimator(
       model_fn=model_function, model_dir=flags.model_dir, config=run_config,
       params={
           'resnet_size': flags.resnet_size,
           'data_format': flags.data_format,
           'batch_size': flags.batch_size,
-          'multi_gpu': flags.multi_gpu,
           'version': flags.version,
       })
 
   if flags.benchmark_log_dir is not None:
     benchmark_logger = logger.BenchmarkLogger(flags.benchmark_log_dir)
-    benchmark_logger.log_run_info("resnet")
+    benchmark_logger.log_run_info('resnet')
```

Comment:
I swear I've changed this in about 5 branches now.

Reply:
No kidding.